### PR TITLE
Improve documentation: Add Quickstart section

### PR DESCRIPTION
**Pull request by malteneuss**
The existing Readme is missing a minimal example how to get up and running with an existing nixpkgs and this poetry2nix repo. Let's fix that by adding non-flake and flake examples

@adisbladis 
Feel free to modify, request changes or remove this PR.

[Contribution](README.md#contributing) checklist (recommended but not always applicable/required):

- [ ] There's an _[automated test](tests/default.nix)_ for this change
- [ ] Commit messages or code include _references to related issues or PRs_ (including third parties)
- [x] Commit messages are _[conventional](https://www.conventionalcommits.org/)_ - examples from [the log](https://github.com/nix-community/poetry2nix/commits/master) include "feat: add changelog files to fixup hook", "fix(contourpy): allow wheel usage", and "test: add sqlalchemy2 test"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,109 @@ _poetry2nix_ turns [Poetry](https://python-poetry.org/) projects into Nix deriva
...
+
+Hint:
+This example assumes that `nixpkgs` and `poetry2nix` are managed and pinned by
+the handy [niv tool](https://github.com/nmattia/niv). In your terminal just run:
+
+```shell
+nix-shell -p niv
+niv init
+niv add nix-community/poetry2nix
+```
```
Comment (das-g) [inline README.md:54]: Move the assumption about `niv` to before `nix-build default.nix`, as without the `niv` setup the build will result in an error about `nix/sources.nix` missing.
````suggestion
Hint:
This example assumes that `nixpkgs` and `poetry2nix` are managed and pinned by
the handy [niv tool](https://github.com/nmattia/niv). In your terminal just run:

```shell
nix-shell -p niv
niv init
niv add nix-community/poetry2nix
```

You can then build your Python application with Nix by running:

```shell
nix-build default.nix
```

Finally, you can run your Python application from the new `./result` symlinked folder:

```shell
# replace <script> with the name in the [tool.poetry.scripts] section of your pyproject.toml
./result/bin/<script>
```
````

File: README.md [hunk 2]
```diff
@@ -6,6 +6,109 @@ _poetry2nix_ turns [Poetry](https://python-poetry.org/) projects into Nix deriva
...
+
+You can then (build and) run your Python app with
+
+```shell
+nix run .
+```
+
+A larger real-world setup can be found in [./templates/app/flake.nix](./templates/app/flake.nix).
+This example is also exported as a flake template so thath you can start your _poetry2nix_ project
```
Comment (das-g) [inline README.md:96]:
```suggestion
This example is also exported as a flake template so that you can start your _poetry2nix_ project
```

**Review by das-g — CHANGES_REQUESTED** [review]

**Comment by das-g** [comment]
To test these instructions, I tried the following in an existing Poetry project:

1. create `./default.nix` with content
   ```nix
   let
     sources = import ./nix/sources.nix;
     pkgs = import sources.nixpkgs { };
     # Let all API attributes like "poetry2nix.mkPoetryApplication" 
     # use the packages and versions (python3, poetry etc.) from our pinned nixpkgs above
     # under the hood:
     poetry2nix = import sources.poetry2nix { inherit pkgs; };
     myPythonApp = poetry2nix.mkPoetryApplication { projectDir = ./.; };
   in
   myPythonApp
   ```
2. run
   ```shell
   nix-shell -p niv
   niv init
   niv add nix-community/poetry2nix
   ```
   (resulting files: [nix/sources.json](https://github.com/nix-community/poetry2nix/files/15062503/sources.json) & [nix/sources.nix](https://github.com/nix-community/poetry2nix/files/15062507/sources.nix.txt))
3. run `nix-build default.nix`

This gave me the error message
```
error:
       … while evaluating the attribute 'drvPath'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/customisation.nix:228:7:

          227|     in commonAttrs // {
          228|       drvPath = assert condition; drv.drvPath;
             |       ^
          229|       outPath = assert condition; drv.outPath;

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: function 'anonymous lambda' called with unexpected argument 'flit-core'

       at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/development/python-modules/wheel/default.nix:1:1:

            1| { lib
             | ^
            2| , buildPythonPackage
```

<details><summary>output of <code>nix-build default.nix --show-trace</code> (click to expand):</summary>

```
error:
       … while evaluating the attribute 'drvPath'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/customisation.nix:228:7:

          227|     in commonAttrs // {
          228|       drvPath = assert condition; drv.drvPath;
             |       ^
          229|       outPath = assert condition; drv.outPath;

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'python3.10-automatisierte-qualit-tskontrolle-von-websiteverweisen-0.1.0'
         whose name attribute is located at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:302:7

       … while evaluating attribute 'buildInputs' of derivation 'python3.10-automatisierte-qualit-tskontrolle-von-websiteverweisen-0.1.0'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:349:7:

          348|       depsHostHost                = lib.elemAt (lib.elemAt dependencies 1) 0;
          349|       buildInputs                 = lib.elemAt (lib.elemAt dependencies 1) 1;
             |       ^
          350|       depsTargetTarget            = lib.elemAt (lib.elemAt dependencies 2) 0;

       … while evaluating derivation 'python3.10-poetry-core-1.5.1'
         whose name attribute is located at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:302:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.10-poetry-core-1.5.1'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:346:7:

          345|       depsBuildBuild              = lib.elemAt (lib.elemAt dependencies 0) 0;
          346|       nativeBuildInputs           = lib.elemAt (lib.elemAt dependencies 0) 1;
             |       ^
          347|       depsBuildTarget             = lib.elemAt (lib.elemAt dependencies 0) 2;

       … while evaluating derivation 'python-catch-conflicts-hook'
         whose name attribute is located at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:302:7

       … while evaluating attribute 'setuptools' of derivation 'python-catch-conflicts-hook'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/development/interpreters/python/hooks/default.nix:87:16:

           86|       substitutions = {
           87|         inherit pythonInterpreter pythonSitePackages setuptools;
             |                ^
           88|         catchConflicts=../catch_conflicts/catch_conflicts.py;

       … while evaluating derivation 'python3.10-setuptools-67.4.0'
         whose name attribute is located at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:302:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.10-setuptools-67.4.0'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:346:7:

          345|       depsBuildBuild              = lib.elemAt (lib.elemAt dependencies 0) 0;
          346|       nativeBuildInputs           = lib.elemAt (lib.elemAt dependencies 0) 1;
             |       ^
          347|       depsBuildTarget             = lib.elemAt (lib.elemAt dependencies 0) 2;

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/make-derivation.nix:504:14:

          503|
          504|   validity = checkMeta { inherit meta attrs; };
             |              ^
          505|

       … while calling 'assertValidity'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:393:20:

          392|
          393|   assertValidity = { meta, attrs }: let
             |                    ^
          394|       validity = checkValidity attrs;

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:394:18:

          393|   assertValidity = { meta, attrs }: let
          394|       validity = checkValidity attrs;
             |                  ^
          395|     in validity // {

       … while calling 'checkValidity'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:343:19:

          342|   # Along with a boolean flag for each reason
          343|   checkValidity = attrs:
             |                   ^
          344|     # Check meta attribute types first, to make sure it is always called even when there are other issues

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:362:13:

          361|     # --- Put checks that can be ignored here ---
          362|     else if hasDeniedUnfreeLicense attrs && !(hasAllowlistedLicense attrs) then
             |             ^
          363|       { valid = "no"; reason = "unfree"; errormsg = "has an unfree license (‘${showLicense attrs.meta.license}’)"; }

       … while calling 'hasDeniedUnfreeLicense'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:77:28:

           76|   # `allowUnfreePredicate` function.
           77|   hasDeniedUnfreeLicense = attrs:
             |                            ^
           78|     hasUnfreeLicense attrs &&

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:78:5:

           77|   hasDeniedUnfreeLicense = attrs:
           78|     hasUnfreeLicense attrs &&
             |     ^
           79|     !allowUnfree &&

       … while calling 'hasUnfreeLicense'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:51:22:

           50|
           51|   hasUnfreeLicense = attrs:
             |                      ^
           52|     hasLicense attrs &&

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:53:5:

           52|     hasLicense attrs &&
           53|     isUnfree (lib.lists.toList attrs.meta.license);
             |     ^
           54|

       … while calling 'isUnfree'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:49:14:

           48|
           49|   isUnfree = licenses: lib.lists.any (l: !l.free or true) licenses;
             |              ^
           50|

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:53:15:

           52|     hasLicense attrs &&
           53|     isUnfree (lib.lists.toList attrs.meta.license);
             |               ^
           54|

       … while calling 'toList'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/lists.nix:284:12:

          283|   */
          284|   toList = x: if isList x then x else [x];
             |            ^
          285|

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/trivial.nix:430:7:

          429|     { # TODO: Should we add call-time "type" checking like built in?
          430|       __functor = self: f;
             |       ^
          431|       __functionArgs = args;

       … while calling anonymous lambda

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/customisation.nix:88:32:

           87|       # Re-call the function but with different arguments
           88|       overrideArgs = copyArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                ^
           89|       # Change the result of the function call by applying g to it

       … from call site

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/customisation.nix:88:41:

           87|       # Re-call the function but with different arguments
           88|       overrideArgs = copyArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                         ^
           89|       # Change the result of the function call by applying g to it

       … while calling 'makeOverridable'

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/lib/customisation.nix:78:24:

           77|   */
           78|   makeOverridable = f: origArgs:
             |                        ^
           79|     let

       error: function 'anonymous lambda' called with unexpected argument 'flit-core'

       at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/development/python-modules/wheel/default.nix:1:1:

            1| { lib
             | ^
            2| , buildPythonPackage
```

</details>

Are the versions of nixpkgs and poetry2nix that `niv` gave me incompatible to each other?

**Comment by malteneuss** [comment]
@das-g Your issues more likely have to do with the specific packages in your `pyproject.toml` and their licenses:

```
        python3.10-automatisierte-qualit-tskontrolle-von-websiteverweisen-0.1.0'
        ...

         at /nix/store/pp307nbzkgsd6393zl2i9j4j86z5nz9b-nixpkgs-src/pkgs/stdenv/generic/check-meta.nix:53:5:

           52|     hasLicense attrs &&
           53|     isUnfree (lib.lists.toList attrs.meta.license);
             |     ^
           54|

       … while calling 'isUnfree'
```
Some packages create very weird results.

You can try with a single dependency like `numpy`.

File: test.nix
Comment (cpcloud) [inline test.nix:1]: Is this file needed/used/tested somewhere?

Comment (malteneuss) [reply test.nix:1]: Nope. It was just used by me to play around and committed by accident. It's removed now.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**